### PR TITLE
Refactored cmsis_dap_core functions into a CMSIS_DAP_Protocol class

### PR DESCRIPTION
**Pull request by flit**
- Wrapped up `pyOCD.transport.cmsis_dap_core.dap*` functions into a `CMSIS_DAP_Protocol` class. It takes the interface in its constructor. Function names were changed to remove the common "dap" prefix.
- Updated CMSIS_DAP to create and use a protocol instance.
- Updated Maxim targets to match changes.

### CODE REVIEW

File: pyOCD/transport/cmsis_dap_core.py [hunk 1]
```diff
@@ -74,348 +74,350 @@
...
-    cmd.append(ID_INFO[id_])
-    interface.write(cmd)
-
-    resp = interface.read()
-    if resp[0] != COMMAND_ID['DAP_INFO']:
-        raise ValueError('DAP_INFO response error')
-    
-    if resp[1] == 0:
+## @brief This class implements the CMSIS-DAP wire protocol.
+class CMSIS_DAP_Protocol(object):
```
Comment (c1728p9) [inline pyOCD/transport/cmsis_dap_core.py]: Was this made into a class in preparation for another protocol similar to CMSIS_DAP?  If so you might want to add an interface for it.

**Comment by c1728p9** [comment]
Ran automated test and everything passed.  Speed is also unaffected.

```
------ TEST RESULTS ------
Target         Test           Result         Time

nrf51          Basic Test     Pass           4.73000001907
nrf51          Speed Test     Pass           40.0550000668
nrf51          Cortex Test    Pass           6.4849998951
nrf51          Flash Test     Pass           117.74000001
kl46z          Basic Test     Pass           3.83999991417
kl46z          Speed Test     Pass           13.7760000229
kl46z          Cortex Test    Pass           3.28800010681
kl46z          Flash Test     Pass           55.7669999599
k22f           Basic Test     Pass           3.8069999218
k22f           Speed Test     Pass           21.9130001068
k22f           Cortex Test    Pass           4.37399983406
k22f           Flash Test     Pass           73.7760000229
lpc1768        Basic Test     Pass           30.6729998589
lpc1768        Speed Test     Pass           33.6100001335
lpc1768        Cortex Test    Pass           5.81199979782
lpc1768        Flash Test     Pass           158.524000168
k64f           Basic Test     Pass           5.46700000763
k64f           Speed Test     Pass           49.3469998837
k64f           Cortex Test    Pass           5.91400003433
k64f           Flash Test     Pass           160.676000118
lpc800         Basic Test     Pass           5.09500002861
lpc800         Speed Test     Pass           10.5190000534
lpc800         Cortex Test    Pass           4.64100003242
lpc800         Flash Test     Pass           18.7809998989
lpc11u24       Basic Test     Pass           7.3259999752
lpc11u24       Speed Test     Pass           8.9880001545
lpc11u24       Cortex Test    Pass           4.29499983788
lpc11u24       Flash Test     Pass           20.7260000706

Test Time: 879.94900012
All tests passed
```

File: pyOCD/transport/cmsis_dap_core.py (hunk 1 above)
Comment (flit) [inline pyOCD/transport/cmsis_dap_core.py]: It's more just architectural cleanup. Better to associate the interface (USB) object with the protocol, than require it to be passed into every function.

**Comment by flit** [comment]
Thanks for testing!

File: pyOCD/transport/cmsis_dap_core.py (hunk 1 above)
Comment (c1728p9) [inline pyOCD/transport/cmsis_dap_core.py]: sounds good to me

**Pull request merged**